backend/inspirehep/disambiguation/utils.py

```diff
@@ -129,28 +133,28 @@ def reorder_lit_author_names(lit_author_name, author_name):
     author_first_names_normalized = {
         unidecode(name).lower() for name in author_first_names
     }
+    lit_author_first_names_normalized = {
+        unidecode(name).lower() for name in lit_author_first_names.split(" ")
+    }
 
-    for name_idx, name in enumerate(lit_author_name_tokens):
+    for nb, name in enumerate(lit_author_first_last_name_tokens):
         normalized_name = unidecode(name).lower()
-        if (
+        name_only_in_author_last_names = (
             normalized_name in author_last_names_normalized
-            and name not in lit_author_last_names_reordered
-        ):
-            lit_author_last_names_reordered.append(name)
-            continue
+            and normalized_name not in author_first_names_normalized
+        )
+        no_more_possible_lit_author_first_names_in_author_firsts = (
+            author_first_names_normalized & lit_author_first_names_normalized
+        )
```

Comment:
* I think technically you need a bag (i.e. `collections.Counter`) instead of a set to manage the case of people having more than one repetition like `John John, Smith John`). Not sure it matters in practice though, so this is probably ok, I'll leave it up to you (and you might want to add a comment with a disclaimer if you keep this).
* are you sure about the negation in the naming? if this is true, it looks like there are still possible matching first names.

Reply:
I can't come up with any idea when if there might be a match (when using counter)

Reply:
it's fine, no need to support super-corner cases, this is good enough :+1: